### ISSUE TITLE
On "resolved" and "settled"

### ISSUE THREAD

**Issue by Neo42** [domenic/promises-unwrapping#128]
Just for translation purposes, would love to know if the term "resolve" here means "to make a decision" or "to solve a problem or dispute". And does the term "settle" here mean "to reach a decision or an agreement about something, or to end a disagreement" or "be in a certain state"?

Many non-native speakers like me might not be able to tell the differences. And people can mistranslate words because they are not familiar with the context. These ambiguities can make a big difference. A clear answer would be extremely helpful.

Thanks.

**Comment by domenic** [comment]
I don't think they mean either of those things. They mean the precise technical meaning given in this document. The fact that the terms have some meanings in English is not really relevant to what they mean in a promises context.

**Comment by benjamingr** [comment]
Just my take - Domenic is a (much) more authoritative reference as he was involved in the decision. Here is how I always understood things.

Resolved (from Google Dictionary) - "decide firmly on a course of action." - a resolved promise is a promise whose fate is known either by following another promise (Promise.resolve(someOtherPromise)) or a value/error (Promise.resolve(42), Promise.resolve(Promise.reject(new Error(('boo)))

Settled (from Cambridge, since Google oddly refused to give me a definition) - "to reach and remain at a certain level or in a certain state". A settled promise is a promise whose state will never change again (either because it fulfilled or rejected), a settled promise is necessarily resolved (since its fate is known if it will never change again) but not vice versa.

**Comment by Neo42** [comment]
@domenic Yes, I understand what you mean. The terms imply totally different technical details in the context of promises. But is it possible for us to pick a context (meaning) in plain English for each of the terms? Because what I'm trying to do here is that I want to translate these terms (alone) into Chinese as accurately as possible. Their technical details are important, too, and will be carefully handled separately.

In Chinese, the term "resolved" can be translated into "已解决"("solved"), "已决议"("decided"), or "已解析"("analyzed", in the context of "image resolution"). "Settled" can mean "已敲定"("finalized"), "已确定"("confirmed"), or "已解决"("solved"). And people use these different translations in the context of promises all over the place, even on the MDN web doc. There are loads of different versions of them there. It's kinda messy. It's might be hard for people who are exposed to these concepts for the first time to get an intuitive idea about how these terms are used if they don't get a well-considered translation. So I hope you understand how choosing a context in plain English for both of these terms can help.

**Comment by DHclly** [comment]
> > I don't think they mean either of those things. They mean the precise technical meaning given in this document. The fact that the terms have some meanings in English is not really relevant to what they mean in a promises context.
> 
> Yes, I understand what you mean. The terms imply totally different technical details in the context of promises. But is it possible for us to pick a context (meaning) in plain English for each of the terms? Because what I'm trying to do here is that I want to translate these terms (alone) into Chinese as accurately as possible. Their technical details are important, too, and will be carefully handled separately.
> 
> In Chinese, the term "resolved" can be translated into "已解决"("solved"), "已决议"("decided"), or "已解析"("analyzed", in the context of "resolution"). "Settled" can mean "已决定"("determined"), "已确定"("confirmed"), or "已解决"("solved"). And people use these different translations in the context of promises all over the place, even on the MDN web doc. There are loads of different versions of them there. It's kinda messy. It's might be hard for people who are exposed to these concepts for the first time to get an intuitive idea about how these terms are used if they don't get a well-considered translation. So I hope you understand how choosing a context in plain English for each of these terms can help.

I think if the Promise API is define the function name  in `resolved` 、 `reject`  、`then`  ，then most words in article should use this words，not the same meaning words。and somewhere is duplicate content,it can optimization like below


e.g.

1.
```
A promise p is fulfilled if p.then(f,r) will immediately enqueue a Task to call the function f.
```
change to
```
A promise p is resolved if p.then(resolvedFun, rejectFun) will immediately enqueue a Task to call the function resolvedFun.
```
2.
```
A promise is pending if it is neither fulfilled nor rejected.
```
change to 
```
A promise is pending if it is neither resolved nor rejected.
```
3.
```
A promise said to be settled if it is not pending, i.e. if it is either fulfilled or rejected.
```
change to
```
A promise is not pending, if it is either resolved or rejected.
```
4.
```
A promise is resolved if it is settled or if it has been "locked in" match the state of another promise. Attempting to resolve or reject a resolved promise has no effect. 
```
change to
```
A promise is resolved or if it has been "locked in" match the state of another promise. Attempting to resolve or reject a resolved promise has no effect. 
```

**Comment by Neo42** [comment]
@DHclly I believe `resolved` and `fulfilled` have different technical details and can't be used interchangeably. And `resolved` and `rejected` are not mutually exclusive.

**Comment by benjamingr** [comment]
> @DHclly I believe resolved and fulfilled have different technical details and can't be used interchangeably. And resolved and rejected are not mutually exclusive.

That is correct, resolved and fulfilled mean different things. Not only is "resolved" and "rejected" not mutually exclusive - every rejected promise is in fact resolved.

> In Chinese, the term "resolved" can be translated into "已解决"("solved"), "已决议"("decided"), or "已解析"("analyzed", in the context of "image resolution"). "Settled" can mean "已敲定"("finalized"), "已确定"("confirmed"), or "已解决"("solved"). And people use these different translations in the context of promises all over the place, even on the MDN web doc.

I don't know Chinese (I plan to learn one day!) however I would guess the correct ones are:

 - Resolved -> "已决议"("decided") - since the future of the promise's fate is decided.
 - Settled -> "已敲定"("finalized") - since it will not transition to a different state again.

However, I really don't know Chinese so it's just a guess :]

**Comment by Neo42** [comment]
> I don't know Chinese (I plan to learn one day!) however I would guess the correct ones are:
>
> - Resolved -> "已决议"("decided") - since the future of the promise's fate is decided.
> - Settled -> "已敲定"("finalized") - since it will not transition to a different state again.
>
> However, I really don't know Chinese so it's just a guess :]

@benjamingr Thanks for the clarification. Guess we can all learn something new today :)

@domenic Still, I will wait for your resolution.

**Comment by DHclly** [comment]
> @DHclly I believe `resolved` and `fulfilled` have different technical details and can't be used interchangeably. And `resolved` and `rejected` are not mutually exclusive.

![p1](https://user-images.githubusercontent.com/11781439/96223306-46bf1f00-0fc0-11eb-8969-3fa50304dd5d.png)
![p4](https://user-images.githubusercontent.com/11781439/96223315-49ba0f80-0fc0-11eb-8cf9-9cc8e7d25ff6.png)
![p2](https://user-images.githubusercontent.com/11781439/96223684-e54b8000-0fc0-11eb-800a-2bfd9cc595da.png)
![p5](https://user-images.githubusercontent.com/11781439/96223692-e7154380-0fc0-11eb-94ce-723b18f38120.png)


No, I think the `resolved` and `fulfilled` is the same meaning, you can see the photo ,first is promise realization in es6，next is promise realization in jQuery(call Deferred Object)，first is call `fulfilled` ,and next is `resolved` ，the other state `pending` 、`rejected` is use the same words ,this just  like  `property` and `attribute` ，in chinese  is one mean ,only one word call  **属性**

also , the resolved and settled difference you can look this link: https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise

![image](https://user-images.githubusercontent.com/11781439/96224492-4162d400-0fc2-11eb-820a-8590232e84b0.png)


From MDN

so `resolved` is equal the `fulfilled`  ,but `resolved` is not equal the `settled` :)

**Comment by Neo42** [comment]
@DHclly I hope this [document](https://github.com/domenic/promises-unwrapping/blob/master/docs/states-and-fates.md) can explain everything for you.

**Comment by DHclly** [comment]
> @DHclly I hope this [document](https://github.com/domenic/promises-unwrapping/blob/master/docs/states-and-fates.md) can explain everything for you.

I know this，I think  specification and functional realization not always equal,  for examle ,the Deferred Object in jQuery is resolved  
 the state is call `resloved` , is also is right :)

**Comment by Neo42** [comment]
> I think specification and functional realization not always equal

@DHclly I agree with you on this. But this repo explains things about the spec, right? The jQuery promise implementation might not be completely compliant with the spec. But, after all, the spec is the spec, isn't it?

**Comment by benjamingr** [comment]
@DHclly no - you can resolve a promise without settling it or even set it to rejected.

```js

const pending = new Promise(resolve => {
  resolve(new Promise((resolve2) => {})); // the outer promise is resolved, the inner one is not
});

const p = new Promise(resolve => {
  resolve(Promise.reject(new Error()); // this creates a rejected promise
});

```


I suggest you read my prior message more carefully, you misunderstood the different states here.

Resolved does not mean fulfilled nor settled. Resolved just means "the fate of this promise is known, either because it is settled or because it is following another promise".

That's why it's `Promise.resolve` that assimilates whatever you pass to it and not `Promise.fulfill` for example.

**Comment by Neo42** [comment]
I think I can close this one for now. Thanks, guys.

**Comment by DHclly** [comment]
> @DHclly no - you can resolve a promise without settling it or even set it to rejected.
> 
> ```js
> const pending = new Promise(resolve => {
>   resolve(new Promise((resolve2) => {})); // the outer promise is resolved, the inner one is not
> });
> 
> const p = new Promise(resolve => {
>   resolve(Promise.reject(new Error()); // this creates a rejected promise
> });
> ```
> 
> I suggest you read my prior message more carefully, you misunderstood the different states here.
> 
> Resolved does not mean fulfilled nor settled. Resolved just means "the fate of this promise is known, either because it is settled or because it is following another promise".
> 
> That's why it's `Promise.resolve` that assimilates whatever you pass to it and not `Promise.fulfill` for example.

Thank you ，I'm wrong , you are right, I just use this simplely 。never find it can use like this 👍
